web/forms.py

```
@@ -1971,3 +1975,112 @@ def clean_title(self) -> str:
+class VirtualClassroomForm(forms.ModelForm):
+    """Form for creating and editing virtual classrooms."""
+
+    def __init__(self, *args: object, **kwargs: object) -> None:
+        user = kwargs.pop("user", None)
+        super().__init__(*args, **kwargs)
+        if user:
+            self.fields["course"].queryset = Course.objects.filter(teacher=user)
+
+    class Meta:
+        model = VirtualClassroom
+        fields = ["name", "course", "max_students"]
```

Ruff (0.14.1)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)